orbit/cmd/orbit/orbit.go

```diff
@@ -616,6 +616,31 @@ func main() {
 			Platform:       osqueryHostInfo.Platform,
 		}
 
+		// Get the hardware UUID. We use a temporary RocksDB location in order to guarantee that
+		// we're getting true UUID, not a cached UUID. See
+		// https://github.com/fleetdm/fleet/issues/17934 and
+		// https://github.com/osquery/osquery/issues/7509 for more details.
+
+		oi, err := getHostInfo(osquerydPath, filepath.Join(os.TempDir(), "tmp-db"))
+		if err != nil {
+			return fmt.Errorf("get UUID from temp db: %w", err)
+		}
+
+		if oi.HardwareUUID != orbitHostInfo.HardwareUUID {
+			// Then we have moved to a new physical machine, so we should restart!
+			// Removing the osquery DB should trigger a re-enrollment when fleetd is restarted.
+			if err := os.RemoveAll(filepath.Join(c.String("root-dir"), "osquery.db")); err != nil {
```

Comment:
We should probably use `osqueryDB` instead of `filepath.Join(c.String("root-dir"), "osquery.db")` (line 602) to support deployments with a custom location for osquery's DB.